sources/dogtag/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+/*!
+dogtag is a set of tools that detect the hostname of a bottlerocket server/instance and prints it to stdout.
```

Comment:
this is an extreme nit but I think it'd be clearer to describe `dogtag` as just a "hostname resolver" rather than a "set of tools". I'd also like to see an explanation of how it uses the resolvers here as well. Something like this:


> `dogtag` resolves the hostname of a bottlerocket server/instance. It's used to generate `settings.network.hostname`. To accomplish this, it uses a set of standalone binaries in `/var/bottlerocket/dogtag` that resolve the hostname via different methods.
>
> Currently, bottlerocket ships with two hostname resolver binaries:
>
> * `20-imds` - Fetches hostname from EC2 Instance Metadata Service
> * `10-reverse-dns` - Uses reverse DNS lookup to resolve the hostname
>
> `dogtag` runs the resolvers in `/var/bottlerocket/dogtag` in reverse alphanumerical order until one of them returns a hostname, at which point it will exit early and print the returned hostname to `stdout`.